### PR TITLE
Feat/addressing components with missing shadow parts

### PR DESCRIPTION
**Pull request by bruno-amorim**
### Related Ticket(s)

[ADCMS-6148](https://jsw.ibm.com/browse/ADCMS-6148)

### Description

I closed this PR: https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/12009, in behalf of this new one.
I have added the last round of shadow parts that we've missed through our shadow part tickets.

### CODE REVIEW

File: packages/web-components/src/components/markdown/markdown.ts [hunk 1]
```diff
@@ -55,14 +59,16 @@ class C4DMarkdown extends LitElement {
...
         return `<${prefix}-link href="${href}" size="lg" ${
           title ? `title="${title}"` : ''
-        }>${text}</${prefix}-link>`;
+        } part="link">${text}</${prefix}-link>`;
       },
       list(body, ordered) {
         const tag = `${prefix}-${ordered ? 'ordered' : 'unordered'}-list`;
-        return `<${tag}>${body}</${tag}>`;
+        return `<${tag} part="${
+          ordered ? 'ordered-list' : 'unordered-list'
```
Comment (m4olivei) [inline packages/web-components/src/components/markdown/markdown.ts:67]: Lets add a consistent name to either case so if someone wanted to target regardless of the kind of list, they could.
```suggestion
          ordered ? 'list ordered-list' : 'list unordered-list'
```

File: packages/web-components/src/components/markdown/markdown.ts [hunk 2]
```diff
@@ -25,6 +25,10 @@ const { prefix, stablePrefix: c4dPrefix } = settings;
  * Markdown content.
  *
  * @element c4d-markdown
+ * @csspart link - The custom link element. Usage: `c4d-markdown::part(link)`
+ * @csspart ordered-list - The ordered list element. Usage: `c4d-markdown::part(ordered-list)`
+ * @csspart unordered-list - The unordered list element. Usage: `c4d-markdown::part(unordered-list)`
```
Comment (m4olivei) [inline packages/web-components/src/components/markdown/markdown.ts:30]:
```suggestion
 * @csspart unordered-list - The unordered list element. Usage: `c4d-markdown::part(unordered-list)`
 * @csspart list - The list element. Usage: `c4d-markdown::part(list)`
```

File: packages/web-components/src/components/masthead/left-nav-menu.ts [hunk 3]
```diff
@@ -30,6 +30,7 @@ const { prefix, stablePrefix: c4dPrefix } = settings;
  * @csspart side-nav-submenu-content - The content of the side navigation submenu. Usage: `c4d-left-nav-menu::part(side-nav-submenu-content)`
  * @csspart side-nav-submenu-title - The title of the side navigation submenu. Usage: `c4d-left-nav-menu::part(side-nav-submenu-title)`
  * @csspart side-nav-icon - The icon of the side navigation submenu. Usage: `c4d-left-nav-menu::part(side-nav-icon)`
+ *  * @csspart side-nav-item-container - The container for the side navigation item. Usage: `c4d-left-nav-menu::part(side-nav-item-container)`
```
Comment (m4olivei) [inline packages/web-components/src/components/masthead/left-nav-menu.ts:33]:
```suggestion
 *  @csspart side-nav-item-container - The container for the side navigation item. Usage: `c4d-left-nav-menu::part(side-nav-item-container)`
```

File: packages/web-components/src/components/masthead/masthead-l1.ts [hunk 4]
```diff
@@ -599,7 +606,7 @@ class C4DMastheadL1 extends StableSelectorMixin(LitElement) {
       [`${prefix}--masthead__l1-dropdown-column-wide`]: hasWide && isWide,
     });
     return hasWide
-      ? html`<div class=${classes}>${renderedSections}</div>`
+      ? html`<div class=${classes} part="column-wide">${renderedSections}</div>`
```
Comment (m4olivei) [inline packages/web-components/src/components/masthead/masthead-l1.ts:609]: Let’s call this `dropdown-column` generically, and then conditionally add a wide or narrow modifier, eg.
```suggestion
      ? html`<div class=${classes} part="dropdown-column${hasWide && !isWide ? ' dropdown-column--narrow' :  hasWide && isWide ? ' dropdown-column--wide' : ''}">${renderedSections}</div>`
```
Also update docs accordingly.

File: packages/web-components/src/components/masthead/masthead-profile.ts [hunk 5]
```diff
@@ -27,6 +27,7 @@ const { prefix, stablePrefix: c4dPrefix } = settings;
  *
  * @element c4d-masthead-profile
  * @csspart profile-link - The masthead contact link. Usage: `c4d-masthead-contact::part(profile-link)`
+ * @csspart header-menu-ul - The header menu ul element. Usage: `c4d-masthead-contact::part(header-menu-ul)`
```
Comment (m4olivei) [inline packages/web-components/src/components/masthead/masthead-profile.ts:30]:
```suggestion
 * @csspart header-menu - The header menu element. Usage: `c4d-masthead-contact::part(header-menu)`
```

File: packages/web-components/src/components/masthead/masthead-profile.ts [hunk 6]
```diff
@@ -123,7 +124,10 @@ class C4DMastheadProfile extends HostListenerMixin(
         @click=${handleClick}>
         ${authenticated ? UserOnline20() : User20()}
       </a>
-      <ul class="${prefix}--header__menu" aria-label="${ifDefined(menuLabel)}">
+      <ul
+        class="${prefix}--header__menu"
+        aria-label="${ifDefined(menuLabel)}"
+        part="header-menu-ul">
```
Comment (m4olivei) [inline packages/web-components/src/components/masthead/masthead-profile.ts:130]:
```suggestion
        part="header-menu">
```

File: packages/web-components/src/components/notice-choice/notice-choice.ts [hunk 7]
```diff
@@ -37,6 +37,8 @@ const { prefix, stablePrefix: c4dPrefix } = settings;
  * @csspart checkbox-label-text - The checkbox label text. Usage `c4d-notice-choice::part(checkbox-label-text)`
  * @csspart error - The error message. Usage `c4d-notice-choice::part(error)`
  * @csspart section - A section. Usage `c4d-notice-choice::part(section)`
+ * @csspart span-container - The span container. Usage `c4d-notice-choice::part(span-container)`
```
Comment (m4olivei) [inline packages/web-components/src/components/notice-choice/notice-choice.ts:40]:
```suggestion
 * @csspart container - The container. Usage `c4d-notice-choice::part(container)`
```

File: packages/web-components/src/components/notice-choice/notice-choice.ts [hunk 8]
```diff
@@ -469,7 +471,7 @@ class NoticeChoice extends StableSelectorMixin(LitElement) {
 
   checkBoxTemplate(checkbox, checked, hiddenBox) {
     this._onChange(`${hiddenBox.id}_VALUE`, `NC_HIDDEN_${hiddenBox.value}`);
-    return html`<span>
+    return html`<span part="span-container">
```
Comment (m4olivei) [inline packages/web-components/src/components/notice-choice/notice-choice.ts:474]:
```suggestion
    return html`<span part="container">
```

**Review by m4olivei — CHANGES_REQUESTED** [review]
Few minor changes here and there.

File: packages/web-components/src/components/masthead/masthead-l1.ts [hunk 9]
```diff
@@ -104,6 +104,13 @@ function handleDropdownClose(event: FocusEvent | KeyboardEvent) {
  * @csspart l1-dropdown-item - The dropdown item in the L1 dropdown. Usage: `c4d-masthead-l1::part(l1-dropdown-item)`
  * @csspart l1-dropdown-login-link - The login link in the L1 dropdown. Usage: `c4d-masthead-l1::part(l1-dropdown-login-link)`
  * @csspart l1-dropdown-links - A group of l1 dropdown links. Usage: `c4d-masthead-l1::part(l1-dropdown-links)`
+ * @csspart dropdown-item-li - The li element for the dropdown item Usage: `c4d-masthead-l1::part(dropdown-item-li)`
+ * @csspart dropdown-login-link-li - The li element for the login link Usage: `c4d-masthead-l1::part(dropdown-login-link-li)`
+ * @csspart cta-li - The li element for the cta Usage: `c4d-masthead-l1::part(cta-li)`
+ * @csspart dropdown-item-link-li - The li element for the dropdown item link Usage: `c4d-masthead-l1::part(dropdown-item-link-li)`
+ * @csspart dropdown-item-span-li - The li element for the dropdown item span Usage: `c4d-masthead-l1::part(dropdown-item-span-li)`
+ * @csspart dropdown-item-button-li - The li element for the dropdown item button Usage: `c4d-masthead-l1::part(dropdown-item-button-li)`
+ * @csspart column-wide - The column wide container Usage: `c4d-masthead-l1::part(column-wide)`
```
Comment (m4olivei) [inline packages/web-components/src/components/masthead/masthead-l1.ts:113]: Need to update the documention for the new naming.

**Review by m4olivei — CHANGES_REQUESTED** [review]
Looking good! Just one missing change for documentation.

**Review by m4olivei — APPROVED** [review]

**Review by kennylam — APPROVED** [review]

**Pull request merged**